gorapass/src/filter_models.py

```diff
@@ -14,37 +13,37 @@ class ModelFilter:
     FILTER_TYPES = { 'exact', 'partial', 'less_than', 'greater_than' }
 
     @classmethod
-    def filter_hikes(cls, hike_models, filters):
+    def filter_data(cls, data_model, filters):
```

Comment:
These instances of `data_model` should probably be `data_models` to indicate that they're a collection, not a single instance. Starting here through the end of this method.